### PR TITLE
[BugFix] Fix unknown error when query iceberg table with complex type

### LINKED ISSUE #11255: Unknown error when query iceberg table with complex type
### Steps to reproduce the behavior (Required)
1. CREATE TABLE sample (
    id bigint COMMENT 'unique id',
    ss struct<x: double, y: double>,
    data string)
USING iceberg

2. select id from  sample

### Expected behavior (Required)
query correct

### Real behavior (Required)
image

### StarRocks version (Required)
 - You can get the StarRocks version by executing SQL `select current_version()`

### PR DESCRIPTION
**Pull request by Youngwb**
## What type of PR is this：
- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #11255 

## Problem Summary(Required) ：

SR on Iceberg only support primitive type now, idToTypeMapping do not contains the corresponding id for complex types like struct, map etc. so we need to deal with the null value for type

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] I have added user document for my new feature or new function

### CODE REVIEW

**Comment by Youngwb** [comment]
run starrocks_fe_unittest

**Review by caneGuy — DISMISSED** [review]

**Review by dirtysalt — DISMISSED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

File: fe/fe-core/src/main/java/com/starrocks/external/iceberg/cost/IcebergFileStats.java [hunk 1]
```diff
@@ -195,8 +195,10 @@ public static Map<Integer, Object> toMap(Map<Integer, Type.PrimitiveType> idToTy
         }
         ImmutableMap.Builder<Integer, Object> map = ImmutableMap.builder();
         idToMetricMap.forEach((id, value) -> {
-            Type.PrimitiveType type = idToTypeMapping.get(id);
-            map.put(id, Conversions.fromByteBuffer(type, value));
+            if (idToTypeMapping.containsKey(id)) {
```
Comment (DorianZheng) [inline fe/fe-core/src/main/java/com/starrocks/external/iceberg/cost/IcebergFileStats.java:198]: plz annotate it

Comment (GavinMar) [inline fe/fe-core/src/main/java/com/starrocks/external/iceberg/cost/IcebergFileStats.java:198]: The `containsKey` and `get` will query map repeatedly, so why not `get` firstly and then check the return value (whether is null) ?

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 3 / 3 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/external/iceberg/cost/IcebergFileStats.java|3|3|100.00%|[]|

**Review by DorianZheng — DISMISSED** [review]
LGTM

**Comment by DorianZheng** [comment]
plz cp to branch-2.4

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

File: fe/fe-core/src/main/java/com/starrocks/external/iceberg/cost/IcebergFileStats.java (hunk 1 above)
Comment (Youngwb) [reply fe/fe-core/src/main/java/com/starrocks/external/iceberg/cost/IcebergFileStats.java]: ok

**Review by DorianZheng — APPROVED** [review]
LGTM

**Comment by mofeiatwork** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 2 / 2 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/external/iceberg/cost/IcebergFileStats.java|2|2|100.00%|[]|

**Review by dirtysalt — APPROVED** [review]

**Review by gengjun-git — APPROVED** [review]

**Comment by Youngwb** [comment]
@Mergifyio backport branch-2.4

**Pull request merged**